npm/vue2/src/index.ts

```diff
@@ -410,6 +412,17 @@ export const mount = (
 
     Cypress.vue = VTUWrapper.vm
     Cypress.vueWrapper = VTUWrapper
+  }).wait(0, { log: false }).then(() => {
+    if (optionsOrProps.log !== false) {
+      const observer = new MutationObserver(() => callbackFunc)
```

Comment:
Cano you explain the benefit of using `MutationObserver` over something like `Vue.nextTick` here? The observer seems complex, and to add it I think we would need more safety around its use. For example, what happens if a component throws an error before rendering? Is the observer ever disconnected?

Same comment for React, I was a bit hesitant but we could use [act](https://reactjs.org/docs/test-utils.html) to let the component stabilize.

Reply:
I believe the implementation using the `MutationObserver` here was to generalize the solution across different frameworks. As soon as we get the children for the `data-cy-root`, the observer is disconnected as you will see in the `callbackFunc`.
I can see why we can implement framework specific solutions because only `vue2` and `react` are impacted so I'll definitely look into the methods mentioned.

Reply:
As discussed, I have used `Vue.nextTick()` in `npm/vue2` to ensure the hover on mount command functionality for vue2 projects and have used `wait(fn, 0)` in `npm/react`